model.js

```diff
@@ -50,4 +50,33 @@ exports.fetchArticleById = (article_id) => {
     };
     return result.rows[0];
   });
-};
\ No newline at end of file
+};
+
+exports.fetchCommentsByArticleId = (article_id) => {
+  return db
+  .query('SELECT * FROM articles WHERE article_id = $1', [article_id])
+  .then((result) => {
+    if (result.rows.length === 0) {
+      return Promise.reject({
+        status: 404,
+        message: `Not Found: no article found for article_id ${article_id}`
+      });
+    }
+    return db
+    .query(`SELECT 
+      comment_id,
+      votes,
+      created_at,
+      author,
+      body,
+      article_id
+      FROM comments
+      WHERE article_id = $1
+      ORDER BY created_at DESC;`, 
+      [article_id])
+    .then((results) => {
+      return results.rows;
```

Comment:
This works, but it may be worth separating out the article check into a reusable function like Stephen did in the advanced error handling lecture.